File: daysInMonth/daysInMonth.js
/*
 Thirty days hath September,
 April, June, and November;
 All the rest have thirty-one,
 Excepting February alone,
 Which has twenty-eight in line,
 Till leap-year gives it twenty-nine.
 */


// As an Array (super simple)
var numDaysInMonth = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];

// written as just an object (super simple)
var daysInMonth = {
 "28": "February",
 "30": ["September", "April", "June", "November"],
 "31": ["January", "March", "May", "July", "August", "October", "December"]
}
//basic if/else function to determine how many days are in a month

function howManyDaysInMonth(month) {
 //standardize the input
 month = month.toLowerCase();
 if (month == "september" || month == "april" || month == "june" || month == "november") {
  //console.log(30);
  return 30;
 } else if (month === "february") {
  return 28;
 } else {
  return 31;
 }
}
console.log(howManyDaysInMonth("october"));

File: daysInMonth/daysInMonth.py

"""
Thirty days hath September,
April, June, and November;
All the rest have thirty-one,
Excepting February alone,
Which has twenty-eight in line,
Till leap-year gives it twenty-nine.
"""

import sys

# As a list
numDaysInMonth = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]

# written as a dict
daysInMonth = {
    28: ["february"],
    30: ["september", "april", "june", "november"],
    31: ["january", "march", "may", "july", "august", "october", "december"]
}

# basic if/else function to determine how many days are in a month

def howManyDaysInMonth(month):
    # standardize the input
    month = month.lower();
    if (month == "september" or month == "april" or month == "june" or month == "november"):
        return 30
    elif (month == "february"):
        return 28
    else:
        return 31;

#print(howManyDaysInMonth("october"))


# function using the daysInMonth dictionary
def howManyDaysInMonthWithDictionary(month):
    # standardize the input
    month = month.lower();
    days = [31, 30, 28]
    
    for d in days:
        if month in daysInMonth[d]:
            return d
    
    return "Invalid input!"

#print(howManyDaysInMonth("AprIL"))


if __name__ == "__main__":
    """ prints the number of days in the month
    of the first command line argument
    
    ex:
    $python daysInMonth.py september
    30
    """
    print(howManyDaysInMonthWithDictionary(sys.argv[1]))